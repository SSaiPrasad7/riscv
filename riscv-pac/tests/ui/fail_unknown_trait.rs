#[riscv_pac::pac_enum(unsafe CoreInterruptNumber)]
#[derive(Clone, Copy, Debug, PartialEq)]
enum Interrupt {
    I1 = 1,
    I2 = 2,
    I4 = 4,
}

fn main() {}
