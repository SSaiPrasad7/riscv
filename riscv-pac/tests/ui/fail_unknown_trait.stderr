error: Unknown trait name. Expected: 'ExceptionNumber', 'CoreInterruptNumber', 'ExternalInterruptNumber', 'PriorityNumber', or 'HartIdNumber'
 --> tests/ui/fail_unknown_trait.rs:1:30
  |
1 | #[riscv_pac::pac_enum(unsafe InterruptNumber)]
  |                              ^^^^^^^^^^^^^^^
